lib/ex_unit/lib/ex_unit/runner.ex

```diff
@@ -107,28 +118,39 @@ defmodule ExUnit.Runner do
   end
 
   defp run_module(config, module) do
-    test_module = module.__ex_unit__()
-    EM.module_started(config.manager, test_module)
-
-    # Prepare tests, selecting which ones should
-    # run and which ones were skipped.
-    tests = prepare_tests(config, test_module.tests)
-
-    {test_module, pending, finished_tests} =
-      if Enum.all?(tests, & &1.state) do
-        # The pending tests here aren't actually run, so they're already
-        # "finished"
-        {test_module, tests, tests}
-      else
-        spawn_module(config, test_module, tests)
-      end
+    if max_failures_reached?(config.stats_pid, config.max_failures) do
+      # This is reached when spawn_modules launches several test_modules,
+      # and one is invalid, and max_failures is reached
+      send(config.runner_pid, {self(), :max_failures_reached})
+    else
+      test_module = module.__ex_unit__()
+      EM.module_started(config.manager, test_module)
+
+      # Prepare tests, selecting which ones should be run or skipped
+      tests = prepare_tests(config, test_module.tests)
+
+      {test_module, test_results} =
+        if Enum.all?(tests, & &1.state) do
+          # The pending tests here aren't actually run,
+          # so they're marked as "finished".
+          {test_module, %{pending: tests, finished: tests}}
```

Comment:
@eksperimental I like this change, but it generates some noise in this not-so-small PR. 

I understand that while working on big PRs like this one you'll see a lot of points for improvement, but please, try to keep you focused on the main goal of the PR 🙂